MarkArts/kuberc
# review

- add check for services
- add check for deployment selector and envfrom
- check for existence of env
- allow skipping configmap and key refs
- fix broken example
- fix broken example
- add check for pvc
- add back the broken example
- split checks into functions
- add some types
- refactor most functions to return issues
- rename errors to issues
- return issues for deployment or statefullset
- Improve output
- add colors
- normalize output
- Update README.md
- run ci on push
- Update README.md
- add verbose option
- test ingress exist
- extend readme
- add tests
- add check for port of podmonitor
- add port checks for service
- add hpa test
- add basic test for deployment and use structuredClone for making coppies of structs
- add tests for volumes
- add check for configmapvolume key
- add tests for configmap envfrom
- add tests for envfrom secret
- add tests for spec.template.spec.containers[].env[].valueFrom
- fix flags and refdoesnotexist error msg
- allow service to be ignored
- update ci
- extend readme with commands
- remove python reference
- typo in readme
- add help
- extend help
- run on ubuntu runners
- add github example
- update github runner target
- correct job names
- fix linting issues
- add workflow_call
- use an action
- use main for version untill fixed
- try different action location
- add action file
- change if syntax of action
- allow read permissions on the passed input file
- don't hash input files for deno cache:
- set correct restore key
- supress deno diagnostic output
- update example version
- update example in readme
- ignore coverage report
- don't assume every container has a port
